app/helpers/templates_helper.rb

```diff
@@ -46,4 +46,28 @@ def locked_warning(template)
 
     alert(:class => 'alert-warning', :text => warning_text.html_safe)
   end
+
+  def template_input_header(f, template)
+    header = _('Template input')
+    unless template.locked?
+      header += ' ' + remove_child_link('x', f, {:rel => 'twipsy', :'data-title' => _('remove template input'), :'data-placement' => 'left',
+                                                 :class => 'fr badge badge-danger'})
+    end
+    header.html_safe
+  end
+
+  def template_input_types_options(keys = TemplateInput::TYPES.keys)
+    keys.map!(&:to_s)
+    TemplateInput::TYPES.select { |k, _| keys.include?(k.to_s) }.map { |key, name| [ _(name), key ] }
+  end
+
+  def advanced_switch_f(default_text, switch_text)
```

Comment:
Can this go to where all the other form helpers live?

Reply:
ok, also extracted the respective JS part to separate file under webpack so it's isolated from templates